contracts/SolutionSubmitter.sol

```diff
@@ -0,0 +1,73 @@
+pragma solidity ^0.5.0;
+
+import "./BatchExchange.sol";
+import "@openzeppelin/contracts/ownership/Ownable.sol";
+
+interface GasToken {
+    function freeUpTo(uint256 value) external returns (uint256 freed);
+}
+
+/** @title SolutionSubmitter - A simple wrapper contract allowing to submit settlements
+ * to the BatchExchange contract while saving gas via the GST2 or CHI GasToken
+ */
+contract SolutionSubmitter is Ownable {
+    BatchExchange public exchange;
+    GasToken public gasToken;
+    uint256 public gasPriceThreshold;
+
+    constructor(
+        BatchExchange _exchange,
+        GasToken _gasToken,
+        uint256 _gasPriceThreshold
+    ) public {
+        exchange = _exchange;
+        setGasToken(_gasToken);
+        setGasPriceThreshold(_gasPriceThreshold);
+    }
+
+    function setGasToken(GasToken _gasToken) public onlyOwner {
+        gasToken = _gasToken;
+    }
+
+    function setGasPriceThreshold(uint256 _gasPriceThreshold) public onlyOwner {
+        gasPriceThreshold = _gasPriceThreshold;
+    }
+
+    /**
+     * Allow the owner to execute arbitrary code (e.g. transfer out solution fee rewards from BatchExchange)
```

Comment:
nit: These doc-comments should probably follow NatSpec format: https://solidity.readthedocs.io/en/v0.7.4/natspec-format.html
```suggestion
     * @dev Allow the owner to execute arbitrary code (e.g. transfer out solution fee rewards from BatchExchange)
```